src/service/inactive-administrative-cost-service.ts

```diff
@@ -0,0 +1,320 @@
+import { RequestWithToken } from '../middleware/token-middleware';
+import { asBoolean, asNumber } from '../helpers/validators';
+import { PaginationParameters } from '../helpers/pagination';
+import {
+  BaseInactiveAdministrativeCostResponse,
+  UserToInactiveAdministrativeCostResponse,
+} from '../controller/response/inactive-administrative-cost-response';
+import { parseUserToBaseResponse } from '../helpers/revision-to-response';
+
+
+export interface InactiveAdministrativeCostFilterParameters {
+  /**
+   * Filter based on user id
+   */
+  fromId?: number;
+
+  /**
+   * Filter based on inactive administrative cost id
+   */
+  inactiveAdministrativeCostId?: number;
+
+  /**
+   * Filter on notification or fine
+   */
+  notification?: boolean;
+}
+
+export function parseInactiveAdministrativeCostFilterParameters(req: RequestWithToken): InactiveAdministrativeCostFilterParameters {
+  return {
+    fromId: asNumber(req.query.fromId),
+    inactiveAdministrativeCostId: asNumber(req.query.inactiveAdministrativeCostId),
+    notification: asBoolean(req.query.notification),
+  };
+}
+
+export default class InactiveAdministrativeCostService extends WithManager {
+
+  // Calculate the year difference between 2 dates.
+  private static yearDifference(date: Date) : number {
+    const dateDiff = (new Date().getTime() - date.getTime());
+    const ageDate = new Date(dateDiff);
+
+    return Math.abs(ageDate.getUTCFullYear() - 1970);
+  }
+
+  public static toArrayResponse(inactiveAdministrativeCosts: InactiveAdministrativeCost[]): BaseInactiveAdministrativeCostResponse[] {
+    return inactiveAdministrativeCosts.map(inactiveAdministrativeCost => InactiveAdministrativeCostService.asInactiveAdministrativeCostResponse(inactiveAdministrativeCost));
+  }
+
+  /**
+   * Parses an InactiveAdministrativeCost Object to a BaseInactiveAdministrativeCostResponse
+   * @param inactiveAdministrativeCost - The InactiveAdministrativeCost to parse
+   */
+  public static asInactiveAdministrativeCostResponse(inactiveAdministrativeCost: InactiveAdministrativeCost): BaseInactiveAdministrativeCostResponse {
+    return {
+      id: inactiveAdministrativeCost.id,
+      createdAt: inactiveAdministrativeCost.createdAt.toISOString(),
+      updatedAt: inactiveAdministrativeCost.updatedAt.toISOString(),
+      from: parseUserToBaseResponse(inactiveAdministrativeCost.from, false),
+      amount: inactiveAdministrativeCost.amount.toObject(),
+      transfer: inactiveAdministrativeCost.transfer ? TransferService.asTransferResponse(inactiveAdministrativeCost.transfer) : undefined,
+    };
+  }
+
+  /**
+   * Checks which users are eligible for either a notification or a fine.
+   * @param params
+   */
+  public async checkInactiveUsers(params: InactiveAdministrativeCostFilterParameters)
+    : Promise<UserToInactiveAdministrativeCostResponse[]> {
+    const { notification } = params;
+    const differenceDate = notification ? 2 : 3;
+
+    const users = await User.find();
```

Comment:
uhhh, you want to loop over `all` users? why not write this using query builder? 
the set of people who will be found is very low, meaning that this seems a bit much.

https://orkhan.gitbook.io/typeorm/docs/select-query-builder